### ISSUE TITLE
What about adding a whiteboard animation?

### ISSUE THREAD

**Issue by ubaldot** [ManimCommunity/manim#3381]
## Description of proposed feature
Given the popularity of whiteboard explainer videos (see e.g. [this](https://www.youtube.com/watch?v=jDm6lvxjAp0)), I was thinking that it may be a low-hanging fruit to add this kind of animation. 

## Ideas on possible implementation directions
It should be enough to add a new "creation" and "un-creation" classes like Write, Create, etc. 

- The creation class (e.g. HandwriteCreate) should make the writing hand to follow the Bezier curve of the MObject that is being unveiled. The MObject could be for example unveiled with .get_subcurve(0, dalpha) where the updater dalpha belongs to [0,1]. The tip of the pen can follow the get_end() point of the returned subcurve.
- The un-creation class (e.g. HandwriteUncreate) should just randomly move the hand holding the eraser around the Bezier curve resulting from .get_subcurve(0, dalpha) - in this case dalpha go from 1 to 0.

**Comment by uwezi** [comment]
use an updater function and it should be pretty simple to get a moving hand...

**Comment by u7416555** [comment]
I am willing to work on this issue. If I can, What would your advice be to work on this. What will I have to know before to get going with this issue.

**Comment by u7416555** [comment]
I would like to work on this issue. Can the maintainers please let me know if this feature is required for this project?

**Comment by u7416555** [comment]
I am trying to work on this issue for my university assignment by teacher is not accepting to work on it as this issue does not have descriptive requirements. Can the maintainers please provide proper guildlines/requirements for this issue please

**Comment by ubaldot** [comment]
What is not clear about the requirements? Just ask! Gathering requirement is part of SW development process.  :) 

Here is an [example](https://www.youtube.com/watch?v=BEAewCiQuDg&list=PLU9ZMRuUFbug1vjfbOnxYeT5km7hZbsCC&index=5) of whiteboard animations. 

All we need is a hand that trace the drawing of VMobjects as they appear to screen.

**Comment by ubaldot** [comment]
True, but [evidence shows](https://www.sciencedirect.com/science/article/pii/S240584402300436X) that whiteboard animations are very effective indeed.

But fine. I will use a different tool for that purpose (like Doodly, etc.).

**Comment by ubaldot** [comment]
> xD Excuse me, what do you want to say with this, which means Laugh Out Loud?

**Comment by MrDiver** [comment]
No, but at this point with your style of argumentation and not really being constructive in many ways i do not really care if i put an "xD" into my answers.

Because it is not my responsibility to take care of what you want. It is my responsibility to find solutions/implementations to ideas which will not result in added weight to the dev process in the future or are just really low priority in the current situation especially if it's not a bug.

Because as already stated in our documentation we are going through a big rewrite at the moment.

And i do not see why we should spent time on implementing this because there is not much capacity if you have looked at the number of issues that have piled up.

So i would rather spent my time with a well written PR or a constructive discussion which will benefit the library in the long run than discussing with you about things that are either not possible or you can just implement them by yourself and not be offended by a dev saying it's not something that is of interest right now.

So in many ways thanks for your ideas but just throwing "evidence" around when i said to you, you can implement it by yourself is just not constructive in any way, you already layed out how you might implement something like this.

So if you want this to exist, then create it and if people like it and actually want it then it will be at some point part of manim.

But please think about the scenario that we are a small team keeping this project alive and that you should stay constructive.

Because it is really annoying to have a discussion about something that could've already been done and made a PR such that if someone was interested in this it would maybe be part of manim.

I hope for the next discussion we can maybe be constructive for once.
